wandb/sdk/launch/docker.py

```diff
@@ -185,6 +197,30 @@ def build_docker_image_if_needed(
         name_line=name_line,
     )
 
+    # add env vars
+    if _is_wandb_local_uri(api.settings("base_url")) and sys.platform == "darwin":
+        _, _, port = _, _, port = api.settings("base_url").split(":")
+        base_url = "http://host.docker.internal:{}".format(port)
+    elif _is_wandb_dev_uri(api.settings("base_url")):
+        base_url = "http://host.docker.internal:9002"
+    else:
+        base_url = api.settings("base_url")
+    env_vars = '\n'.join([
+        f"ENV WANDB_BASE_URL={base_url}",
+        f"ENV WANDB_API_KEY={api.api_key}",
+        f"ENV WANDB_PROJECT={launch_project.target_project}",
+        f"ENV WANDB_ENTITY={launch_project.target_entity}",
+        f"ENV WANDB_LAUNCH={True}",
+        f"ENV WANDB_LAUNCH_CONFIG_PATH={os.path.join(workdir,_project_spec.DEFAULT_LAUNCH_METADATA_PATH)}",
+        f"ENV WANDB_RUN_ID={launch_project.run_id or None}",
+        f"ENV WANDB_DOCKER={launch_project.docker_image}",
+    ])
+    dockerfile_contents += env_vars + "\n"
+
+    if runner_type == 'gcp-vertex':
+        # supply the entrypoint via the dockerfile since we don't `docker run`
+        dockerfile_contents += 'ENTRYPOINT ["python", "{entrypoint}"]\n'.format(entrypoint=launch_project.get_single_entry_point().name)
```

Comment:
This needs to have the override args added to it. See our function `get_entry_point_command`

Reply:
gcp takes overrides as a separate arg at the job construction level so i thought i'd keep it there